android/src/main/java/com/appspector/flutter/MainAppSpectorHandler.java

```diff
@@ -0,0 +1,243 @@
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import io.flutter.plugin.common.BinaryMessenger;
+import io.flutter.plugin.common.MethodCall;
+import io.flutter.plugin.common.MethodChannel;
+
+class MainAppSpectorHandler implements MethodChannel.MethodCallHandler {
+
+    private final Application application;
+    @SuppressWarnings({"FieldCanBeLocal", "unused"})
+    private final MethodChannel mainChannel;
+    private final EventReceiver eventReceiver;
+    private final RequestSender requestSender;
+    private final SessionUrlListener sessionUrlListener;
+    private final Map<String, MonitorInitializer> monitorInitializerMap;
+
+    private MainAppSpectorHandler(Application application,
+                                  MethodChannel mainChannel,
+                                  SessionUrlListener sessionUrlListener,
+                                  EventReceiver eventReceiver,
+                                  RequestSender requestSender) {
+        this.application = application;
+        this.mainChannel = mainChannel;
+        this.eventReceiver = eventReceiver;
+        this.requestSender = requestSender;
+        this.sessionUrlListener = sessionUrlListener;
+        this.monitorInitializerMap = createMonitorInitializerMap();
+        registerEvents(eventReceiver);
+    }
+
+    private void registerEvents(EventReceiver eventReceiver) {
+        eventReceiver.registerEventHandler(new HttpRequestEventHandler());
+        eventReceiver.registerEventHandler(new HttpResponseEventHandler());
+        eventReceiver.registerEventHandler(new LogEventHandler());
+    }
+
+    void register() {
+        mainChannel.setMethodCallHandler(this);
+    }
+
+    void unregister() {
+        mainChannel.setMethodCallHandler(null);
+        eventReceiver.unsubscribe();
+    }
+
+    @Override
+    public void onMethodCall(@NonNull MethodCall call, @NonNull MethodChannel.Result result) {
+        switch (call.method) {
+            case "run":
+                initAppSpector(result,
+                        call.argument("apiKey"),
+                        call.argument("metadata"),
+                        call.argument("enabledMonitors")
+                );
+                break;
+            case "stop":
+                stopSdk(result);
+                break;
+            case "start":
+                startSdk(result);
+                break;
+            case "isStarted":
+                checkSdkStarted(result);
+                break;
+            case "setMetadata":
+                setMetadata(call.argument("key"), call.argument("value"), result);
+                break;
+            case "removeMetadata":
+                removeMetadata(call.argument("key"), result);
+                break;
+            default:
+                result.notImplemented();
+        }
+    }
```

Comment:
**Suggestion:** The switch statement in the onMethodCall function could be replaced with a map of function references for better readability and maintainability.
```suggestion
        Map<String, Runnable> methodMap = new HashMap<>();
        methodMap.put("run", () -> initAppSpector(result, call.argument("apiKey"), call.argument("metadata"), call.argument("enabledMonitors")));
        methodMap.put("stop", () -> stopSdk(result));
        methodMap.put("start", () -> startSdk(result));
        methodMap.put("isStarted", () -> checkSdkStarted(result));
        methodMap.put("setMetadata", () -> setMetadata(call.argument("key"), call.argument("value"), result));
        methodMap.put("removeMetadata", () -> removeMetadata(call.argument("key"), result));

        Runnable method = methodMap.get(call.method);
        if (method != null) {
          method.run();
        } else {
          result.notImplemented();
        }
```